ten/four_nieoh.py

```diff
@@ -0,0 +1,21 @@
+import heapq
+
+def distance(x, y, z):
+    return -(x**2 + y **2 + z**2)
+
+def closest_stars(stars, k):
+    H = [(distance(*star), star) for star in stars[:k]]
+    heapq.heapify(H)
+    i = 0
+    ret = []
+    for star in stars:
+        if i < k:
```

Comment:
I would have just started with an empty heap instead of special casing like this.